Libraries/LibGUI/TreeView.cpp

```diff
@@ -110,13 +110,16 @@ void TreeView::doubleclick_event(MouseEvent& event)
 
 void TreeView::set_open_state_of_all_in_subtree(const ModelIndex& root, bool open)
 {
-    if (root.is_valid())
+    if (root.is_valid()) {
         ensure_metadata_for_index(root).open = open;
+        if (model()->row_count(root)) {
+            if (on_toggle)
+                on_toggle(root, open);
+        }
+    }
```

Comment:
This is entirely unrelated to the other change, so please put it in a separate commit :)

Reply:
Righto!